src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const { type, keysToRemove, extraData } of actions) {
+    switch (type) {
+      case 'addProperties':
+        currentState = { ...currentState, ...extraData };
+        break;
+
+      case 'removeProperties':
+        for (const prop of keysToRemove) {
+          delete currentState[prop];
+        }
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+
+      default:
+        break;
```

Comment:
Thats good!

However, try also performing some meaningful action in such cases.
For instance - use logging to report that something went wrong:
```suggestion
      default:
         console.warn('Unexpected action type: ' + type)
      break;
```